service/tools_post.go

```diff
@@ -0,0 +1,42 @@
+package service
+
+import (
+	"encoding/json"
+	"github.com/bitrise-io/api-utils/httpresponse"
+	"github.com/bitrise-io/bitrise-step-analytics/models"
+	"github.com/pkg/errors"
+	"net/http"
+)
+
+// ToolsPostHandler ...
+func ToolsPostHandler(w http.ResponseWriter, r *http.Request) error {
+	var toolAnalytics models.ToolAnalytics
+	defer httpresponse.RequestBodyCloseWithErrorLog(r)
+	if err := json.NewDecoder(r.Body).Decode(&toolAnalytics); err != nil {
+		return httpresponse.RespondWithBadRequestError(w, "Invalid request body, JSON decode failed")
+	}
+
+	if toolAnalytics.BuildSlug == "" {
+		return httpresponse.RespondWithBadRequestError(w, "Invalid request body, please provide build_slug")
+	}
+	if len(toolAnalytics.ToolUsage) < 1 {
+		return httpresponse.RespondWithBadRequestError(w, "Invalid request body, please provide at least one tool_usage")
+	}
+	for _, toolUsage := range toolAnalytics.ToolUsage {
+		if toolUsage.Name == "" {
```

Comment:
What about the other properties `Version` and `FreshInstall`?

Reply:
The default value is OK for them

Reply:
Ok, I see, I agree with `Version` being empty. But are we correctly assuming that if we cannot determine if a tool is freshly installed, we say it is updated?

Reply:
Yes. Our tool will be filling the params right. Other users of the endpoint will have to fill the field properly. I think false is a safe default here.